### PR TITLE
Added conditional to skip render model if model in doc root is not component

### LINKED ISSUE #3117: Add a test for `Document.add_root(not_a_component)`
For use with `get_model_by_name` for example we were talking about having data sources as roots. (See https://github.com/bokeh/bokeh/pull/3112 ). The old 0.10 document only allowed Component subtypes at the root; we may still have checks for that, or `embed.coffee` may try to render all roots even if they aren't "renderable". So we should test (and fix if necessary) having a data source or other non-renderable model as a root.

### PR DESCRIPTION
**Pull request by canavandl**
issues: fixes #3117 

stops ``render_model`` if model doesn't have ``default_view``. Handles the case where a model in the document root is not a component (i.e. is a datasource).

### CODE REVIEW

**Comment by canavandl** [comment]
Thinking about it - I'm going to add a warn explaining the the model doesn't have a default_view method and won't be rendered instead of killing the render silently.

**Comment by bryevdv** [comment]
I think this is good but what about adding checks on the python side? If there isn't already one, we could make a base class for "viewable" things that `add_root` could check for. We could also add a BokehJS unit test to ensure that any "viewable" thing also defines a `default_view`.

**Comment by canavandl** [comment]
I think it might be possible to use the `Renderer` base class as a class for 'viewable' things to test for.

I'll also add a BokehJS unit test to look for `default_view`.

**Comment by canavandl** [comment]
Made Document.add_root method require Components (plots, layouts, widgets) as arguments with test.

**Comment by bryevdv** [comment]
Should add a line in release notes about moving to `bokeh.layouts` (but note still in `bokeh.models`)

**Comment by canavandl** [comment]
I think I already did, but I'll tweak the working real quick.

File: sphinx/source/docs/releases/0.11.1.rst [hunk 1]
```diff
@@ -6,3 +6,5 @@ improvements from the Jan 2016 0.11.0 release.
 
 Migration notes:
 * Range update callbacks now return the range object as the cb_obj (not the plot)
+* Layouts (HBox, VBox, VBoxForm) have been moved from bokeh.models.widgets to
+  bokeh.models.layouts
```
Comment (bryevdv) [inline sphinx/source/docs/releases/0.11.1.rst]: Yah I would just add "(but continue to be importable from bokeh.models, as before)"

**Comment by birdsarah** [comment]
This is great but there is currently no ability to set arbitrary attributes on a document/session the "vars namespace" as @bryevdv calls it.

I have been adding ColumnDataSources to the root as a workaround for this missing feature.

As a result, I'd be grateful if this PR didn't get merged until the "vars namespace" is available.

I believe the issue is: https://github.com/bokeh/bokeh/issues/3349

**Comment by canavandl** [comment]
superseded by #3703

**Pull request closed without merging**